packages/block-library/src/spacer/controls.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { InspectorControls, useSetting } from '@wordpress/block-editor';
+import {
+	BaseControl,
+	PanelBody,
+	__experimentalUseCustomUnits as useCustomUnits,
+	__experimentalUnitControl as UnitControl,
+} from '@wordpress/components';
+import { useInstanceId } from '@wordpress/compose';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	MIN_SPACER_HEIGHT,
+	MAX_SPACER_HEIGHT,
+	MIN_SPACER_WIDTH,
+	MAX_SPACER_WIDTH,
+} from './edit';
+
+function DimensionInput( {
+	label,
+	onChange,
+	onUnitChange,
+	unit = 'px',
+	value = '',
+	max,
+	min,
+} ) {
+	const [ temporaryInput, setTemporaryInput ] = useState( null );
+
+	const instanceId = useInstanceId( UnitControl );
+	const inputId = `block-spacer-height-input-${ instanceId }`;
```

Comment:
A prefix can be passed as a second argument for `useInstanceId`.

Reply:
Good to know! Updated.